ladder

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+
+import argparse
+from argparse import RawTextHelpFormatter
+import os
+import sys
+import subprocess
+
+
+def test(args, runner_args):
+    cargo_args = ["cargo", "test"]
+    env = os.environ.copy()
+    if args.sys:
+        cargo_args.append("--test")
+        cargo_args.append("system_tests")
+    cargo_args.append("--")
+    cargo_args.append("--test-threads")
+    cargo_args.append(str(args.threads))
+    for runner_arg in unknown:
+        cargo_args.append(runner_arg)
+    cargo_args.append("--")
+    if args.filter:
+        cargo_args.append(args.filter)
+
+    if args.bless:
+        env["BLESS"] = "1"
+
+    process = subprocess.run(cargo_args, env=env)
+    sys.exit(process.returncode)
+
+
+def bench(args, _unknown):
+    os.chdir("libslide")
+    cargo_args = ["cargo", "bench"]
+    if args.filter:
+        cargo_args.append(args.filter)
+    cargo_args.append("--features")
+    cargo_args.append("benchmark-internals")
+
+    process = subprocess.run(cargo_args)
+    sys.exit(process.returncode)
+
+
+def lf(args, _unknown):
+    process = subprocess.run(["./scripts/check", "lf"])
+    sys.exit(process.returncode)
+
+
+parser = argparse.ArgumentParser(
+    prog="ladder",
+    description="ladder: the slide development tool.",
+    formatter_class=RawTextHelpFormatter
+)
+sub_parsers = parser.add_subparsers(help="ladder subcommands")
+
+test_parser = sub_parsers.add_parser(
+    "test",
+    help="Run slide tests",
+    formatter_class=RawTextHelpFormatter,
+    description="""
+Runs slide tests using cargo. Arguments to the underlying "cargo test" runners
+can be passed as extra flags on this command.
+
+This command accepts an optional positional argument that serves as a test
+filter. If no filter is specified, all tests are run.
+
+To run only system tests, pass --sys.
+
+Examples:
+
+> ladder test                  # runs all tests
+> ladder test --sys --green    # runs all system tests and accepts their output
+> ladder test --quiet          # runs `cargo test` in quiet mode
+> ladder test add              # only runs tests with "add" in their name
```

Comment:
I am assuming quiet and green aren't implemented yet or are they a predefined cargo arg